--- a/libpng-1.0.9/pngwutil.c
+++ b/libpng-1.0.9/pngwutil.c
@@ -343,9 +343,10 @@ png_write_compressed_data_out(png_struct
       png_free(png_ptr, comp->output_ptr[i]);
       comp->output_ptr[i]=NULL;
    }
-   if (comp->max_output_ptr != 0)
+   if (comp->max_output_ptr != 0) {
       png_free(png_ptr, comp->output_ptr);
       comp->output_ptr=NULL;
+   }
    /* write anything left in zbuf */
    if (png_ptr->zstream.avail_out < (png_uint_32)png_ptr->zbuf_size)
       png_write_chunk_data(png_ptr, png_ptr->zbuf,
